sound/soc/sof/topology.c

```diff
@@ -2017,6 +2042,60 @@ static int sof_get_control_data(struct snd_soc_component *scomp,
 	return 0;
 }
 
+/*
+ * Sof mux/demux component
+ */
+static int sof_process_load_mux(struct snd_soc_component *scomp,
+				struct sof_ipc_comp_process *process,
+				struct snd_soc_tplg_private *private)
+{
+	struct sof_mux_config *mc = (struct sof_mux_config *)&process->data;
+	struct mux_stream_data *msd = (struct mux_stream_data *)
+		((u8 *)process->data + sizeof(struct sof_mux_config));
+	struct snd_soc_tplg_vendor_array *varray;
+	int pos = 0;
+	int ret;
+	int i;
+
+	/* parse mux config array */
+	ret = sof_parse_tokens(scomp, mc, mux_tokens,
+			       ARRAY_SIZE(mux_tokens), private->array,
+			       le32_to_cpu(private->size));
+	if (ret < 0) {
+		dev_err(scomp->dev, "error: parse mux tokens failed %d\n",
+			private->size);
+		return ret;
+	}
+
+	/* parse mux stream config arrays */
+	for (i = 0; i < mc->num_streams; i++) {
```

Comment:
nitpick: you could add msd++ to the loop statement:
```
for (i = 0; i < mc->num_streams; i++, msd++) {
```
to make it more obvious, that each loop iteration advances `msd`

Reply:
ok, actually added even some more, hope it doesn't get confusing :)